lib/secrets.nix

```diff
@@ -0,0 +1,75 @@
+{ lib }:
+with lib.strings;
+with lib.types;
+with lib.options;
+rec {
+
+
+  # A type for paths containing secrets
+  secretPath = mkOptionType {
+    name = "secret path";
+    check = x: path.check x && (
+      # Secrets shouldn't be path literals because those could be copied into the store
+      # Secrets also shouldn't be in /nix/store through other means
+      builtins.typeOf x != "path" && ! hasPrefix builtins.storeDir (toString x)
+      # Unless this is explicitly wanted with secretInNixStore
+      || x._secretInNixStore or false);
+    checkFailedMessage = x: optionalString (path.check x) "If you want to import the secret into the globally readable Nix store, use lib.secretInNixStore on the value.";
```

Comment:
Sounds good (once the docs exist)

Reply:
Instead of `secretInNixStore`, which may still be interpreted as being secure/encrypted, I would call it something more ominous like `unprotectedSecretPath`.